<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from inmoov_right_hand.urdf.xacro   | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="inmoov_hand" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.0001" radius="0.26"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.055"/>
      <material name="invisible">
        <color rgba="0.3 0.3 0.3 0"/>
      </material>
    </visual>
  </link>
  <link name="right_hand">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_hand.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_thumb1">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_thumb5_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="3.1415 0 0" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_thumb2">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/thumb5_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_thumb3">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/thumb5_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_index1">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/index3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_index2">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/index3_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_index3">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/index3_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_middle1">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/middle3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_middle2">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/middle3_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_middle3">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/middle3_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_ring1">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_ring3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_ring2">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/ring3_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_ring3">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/ring3_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_ring4">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/ring3_4.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_pinky1">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_pinky3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_pinky2">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/pinky3_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_pinky3">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/pinky3_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <link name="right_pinky4">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/pinky3_4.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="cframe">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <joint name="right_thumb1" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0" upper="1.57" velocity="0.5"/>
    <origin rpy="3.1415 0.1 0" xyz="-0.029 0 -0.0577"/>
    <mimic joint="right_thumb" multiplier="-0.75" offset="0"/>
    <parent link="right_hand"/>
    <child link="right_thumb1"/>
  </joint>
  <joint name="right_thumb" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.5" velocity="0.5"/>
    <origin rpy="0 2.3 -0.3" xyz="-0.02725 -0.0006 0.014"/>
    <parent link="right_thumb1"/>
    <child link="right_thumb2"/>
  </joint>
  <joint name="right_thumb3" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-1.57" upper="0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0 -0.035"/>
    <mimic joint="right_thumb" multiplier="1" offset="0"/>
    <parent link="right_thumb2"/>
    <child link="right_thumb3"/>
  </joint>
  <joint name="right_index1" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.5" velocity="0.5"/>
    <origin rpy="0 -0.1 3.1415" xyz="-0.0342 -0.0015 -0.119"/>
    <mimic joint="right_index" multiplier="1" offset="0"/>
    <parent link="right_hand"/>
    <child link="right_index1"/>
  </joint>
  <joint name="right_index" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.5" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0.001 -0.03595"/>
    <parent link="right_index1"/>
    <child link="right_index2"/>
  </joint>
  <joint name="right_index3" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.5" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0 -0.024"/>
    <mimic joint="right_index" multiplier="1" offset="0"/>
    <parent link="right_index2"/>
    <child link="right_index3"/>
  </joint>
  <joint name="right_middle1" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.5" velocity="0.5"/>
    <origin rpy="0 0 3.1415" xyz="-0.007 -0.00175 -0.12325"/>
    <mimic joint="right_middle" multiplier="1" offset="0"/>
    <parent link="right_hand"/>
    <child link="right_middle1"/>
  </joint>
  <joint name="right_middle" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.5" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0.000 -0.0389"/>
    <parent link="right_middle1"/>
    <child link="right_middle2"/>
  </joint>
  <joint name="right_middle3" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-1.57" upper="0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0.0005 -0.0259"/>
    <mimic joint="right_middle" multiplier="1" offset="0"/>
    <parent link="right_middle2"/>
    <child link="right_middle3"/>
  </joint>
  <joint name="right_ring1" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.57" upper="0" velocity="0.5"/>
    <origin rpy="0 -0.7 3.1415" xyz="0.00705 0 -0.0794"/>
    <mimic joint="right_ring" multiplier="0.1" offset="0"/>
    <parent link="right_hand"/>
    <child link="right_ring1"/>
  </joint>
  <joint name="right_ring" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.5" velocity="0.5"/>
    <origin rpy="0 0.7775 0" xyz="-0.0351 0.00126 -0.0166"/>
    <parent link="right_ring1"/>
    <child link="right_ring2"/>
  </joint>
  <joint name="right_ring3" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-1.57" upper="0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0.0005 -0.0345"/>
    <mimic joint="right_ring" multiplier="1" offset="0"/>
    <parent link="right_ring2"/>
    <child link="right_ring3"/>
  </joint>
  <joint name="right_ring4" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-1.57" upper="0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0.0004 -0.0229"/>
    <mimic joint="right_ring" multiplier="1" offset="0"/>
    <parent link="right_ring3"/>
    <child link="right_ring4"/>
  </joint>
  <joint name="right_pinky1" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.57" upper="0" velocity="0.5"/>
    <origin rpy="0 -0.7 3.1415" xyz="0.027 0 -0.0555"/>
    <mimic joint="right_pinky" multiplier="0.1" offset="0"/>
    <parent link="right_hand"/>
    <child link="right_pinky1"/>
  </joint>
  <joint name="right_pinky" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.5" velocity="0.5"/>
    <origin rpy="0 0.93 0" xyz="-0.046 -0 -0.0228"/>
    <parent link="right_pinky1"/>
    <child link="right_pinky2"/>
  </joint>
  <joint name="right_pinky3" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-1.57" upper="0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0.0005 -0.031"/>
    <mimic joint="right_pinky" multiplier="1" offset="0"/>
    <parent link="right_pinky2"/>
    <child link="right_pinky3"/>
  </joint>
  <joint name="right_pinky4" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-1.57" upper="0" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0.0004 -0.0208"/>
    <mimic joint="right_pinky" multiplier="1" offset="0"/>
    <parent link="right_pinky3"/>
    <child link="right_pinky4"/>
  </joint>
  <link name="right_handcover">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_cover_hand.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="ccover">
        <color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
  </link>
  <link name="right_cover_handring">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_cover_handring.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="ccover">
        <color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
  </link>
  <link name="right_cover_handpinky">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_cover_handpinky.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="ccover">
        <color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
  </link>
  <link name="right_cover_thumb">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_cover_thumb.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="3.1415 0 0" xyz="0 0 0"/>
      <material name="ccover">
        <color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
  </link>
  <link name="right_cover_index">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_cover_index.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="ccover">
        <color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
  </link>
  <link name="right_cover_middle">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_cover_middle.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="ccover">
        <color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
  </link>
  <link name="right_cover_ring">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_cover_ring.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="ccover">
        <color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
  </link>
  <link name="right_cover_pinky">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/right_cover_pinky.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <material name="ccover">
        <color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
  </link>
  <joint name="right_handcover" type="fixed">
    <origin rpy="0 0 0" xyz="-0.0192 0.009 -0.0977"/>
    <parent link="right_hand"/>
    <child link="right_handcover"/>
  </joint>
  <joint name="right_cover_handring" type="fixed">
    <origin rpy="0 0.7 0" xyz="-0.015 -0.009 -0.0008"/>
    <parent link="right_ring1"/>
    <child link="right_cover_handring"/>
  </joint>
  <joint name="right_cover_handpinky" type="fixed">
    <origin rpy="0 0.7 0" xyz="-0.0205 -0.01 -0.0069"/>
    <parent link="right_pinky1"/>
    <child link="right_cover_handpinky"/>
  </joint>
  <joint name="right_cover_thumb" type="fixed">
    <origin rpy="0 3.1 0" xyz="-0.02125 -0.01 -0.0047"/>
    <parent link="right_thumb1"/>
    <child link="right_cover_thumb"/>
  </joint>
  <joint name="right_cover_index" type="fixed">
    <origin rpy="0.05 0 0" xyz="0 -0.0071 -0.012"/>
    <parent link="right_index1"/>
    <child link="right_cover_index"/>
  </joint>
  <joint name="right_cover_middle" type="fixed">
    <origin rpy="-0.05 0 0" xyz="0 -0.0071 -0.012"/>
    <parent link="right_middle1"/>
    <child link="right_cover_middle"/>
  </joint>
  <joint name="right_cover_ring" type="fixed">
    <origin rpy="-0.05 -0.1 0" xyz="0 -0.005 -0.011"/>
    <parent link="right_ring2"/>
    <child link="right_cover_ring"/>
  </joint>
  <joint name="right_cover_pinky" type="fixed">
    <origin rpy="-0.05 -0.1 0" xyz="0 -0.005 -0.0105"/>
    <parent link="right_pinky2"/>
    <child link="right_cover_pinky"/>
  </joint>
  <joint name="base_to_hand" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.14" upper="0" velocity="0.5"/>
    <origin rpy="0 3.1415 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="right_hand"/>
  </joint>
</robot>

